scripts/gen_binding.py

```diff
@@ -443,6 +443,16 @@ def normalize_arg(arg):
     wrap_args = list(zip(js_args, js_arg_types))
     if methods_only:
         wrap_args = [('this', 'tensor')] + wrap_args
+    js_cast_b8 = ""
+    if valid_js(op) == 'where' and not methods_only:
+        js_cast_b8 = 'if (cond.dtype !== dtype.BoolInt8) cond = cond.astype(dtype.BoolInt8)'
+    elif valid_js(op) == 'where':
+        js_cast_b8 = f"""\
+          if(this.dtype !== dtype.BoolInt8) {{
+            const temp_tensor = this.astype(dtype.BoolInt8)
+            this.update(temp_tensor)
```

Comment:
I'd try to avoid `update` where possible.  It does a bunch of messy stuff (besides being the only non-functional method in the entirely library, it also kicks off a full garbage collection)